bindings/rust/s2n-tls/src/connection.rs

```diff
@@ -1049,6 +1049,42 @@ impl Connection {
     pub fn resumed(&self) -> bool {
         unsafe { s2n_connection_is_session_resumed(self.connection.as_ptr()) == 1 }
     }
+
+    /// Associates an arbitrary application context with the Connection to be later retrieved via
+    /// the [`Self::application_context()`] and [`Self::application_context_mut()`] APIs.
+    ///
+    /// This API will override an existing application context set on the Connection.
+    pub fn set_application_context<T: Send + Sync + 'static>(&mut self, app_context: T) {
+        self.context_mut().app_context = Some(Box::new(app_context));
+    }
+
+    /// Retrieves a reference to the application context associated with the Connection.
+    ///
+    /// To set a context on the connection, use the [`Self::set_application_context()`] API. If an
+    /// application context of the specified type hasn't already been set on the Connection, None
+    /// will be returned.
+    ///
+    /// To retrieve a mutable reference to the context, use [`Self::application_context_mut()`].
```

Comment:
I think putting the comment about returning None after the comment about setting the context makes it look at a glance like you're trying to say that set_application_context will return None. I also think it's wroth being clearer that only one context can be set.
```suggestion
    /// If an application context hasn't already been set on the Connection, None will be returned.
    /// If the application context set on the Connection isn't the requested type, None will be returned.
    ///
    /// To set a context on the connection, use the [`Self::set_application_context()`] API. 
    /// To retrieve a mutable reference to the context, use [`Self::application_context_mut()`].
```